docs/sources/developer/architecture/components/api.md

```diff
@@ -0,0 +1,78 @@
+# API
+
+The Faro API is the primary interface for end-users to interact with Faro. It is the central collector of signals, and
+also exposes helper methods for extensions such as metas.
+
+## Signals
+
+### Events
+
+Events are signals that can be used to record user interactions, such as mouse clicks, page navigation, or client-side
+application events like page transitions.
+
+Methods:
+
+- `pushEvent` - sends an event
+
+### Exceptions
+
+Exceptions are signals that are represented by errors which occurred in an app.
+
+When an exception is sent, the API will pass it through a stacktrace parser to extract the relevant information from it.
+By default, the core package does not contain any stacktrace parsers and they are either provided by wrapper packages
+like `web-sdk` or by the user.
+
+Methods:
+
+- `changeStacktraceParser` - replaces the existing stacktrace parser with a new one
+- `getStacktraceParser` - returns the current stacktrace parser
+- `pushError` - sends an exception
+
+### Logs
+
+Logs are signals which capture events that happen within the app itself. They do not have a business meaning but they
+have a use for operators as a means of troubleshooting or observing historic (or current) application behaviour.
+
+Methods:
+
+- `pushLog` - sends a log
+
+### Measurements
+
+Measurements are signals which can be seen as metrics.
+
+Methods:
+
+- `pushMeasurement` - sends a measurement
+
+### Traces
+
+Traces are a special type of signal which are used to track the behaviour of an app. Because Faro does not have an
+internal tracing mechanism, we rely on OpenTelemetry's tracing mechanism to provide this functionality. However,
+OpenTelemetry is quite a heavy system and we do not want to force users to use it. Therefore, the traces API is
```

Comment:
Give a small explanation of what heavy is (computing power, size)

Reply:
I think this is out of scope for this particular set of docs. This should be part of another set of docs where we describe the `web-tracing` package.